`any` type cause c compile error

### Describe the bug

I don't know how to use the base type `any`....

### Reproduction Steps

any.v
```v
fn g[T](x T) {
        for _ in 0 .. 10 {
                _ := any{}
        }
}

fn main() {
        g[int](1)
}
```

v any.v

### Expected Behavior

compile ok or report a error

### Current Behavior

```
$ v any.v
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/any.01JP2D9T6W6YKSN6GBG2S1W7TY.tmp.c:5623: error: 'any' undeclared
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .

```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.9 ab2eb00.50b716b

### Environment details (OS name and version, etc.)

|V full version      |V 0.4.9 ab2eb00.50b716b
|:-------------------|:-------------------
|OS                  |linux, Ubuntu 24.04.2 LTS
|Processor           |8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-9700 CPU @ 3.00GHz
|Memory              |5.05GB/15.51GB
|                    |
|V executable        |/media/HD/github/kbkpbot/v/v
|V last modified time|2025-03-10 03:04:03
|                    |
|V home dir          |OK, value: /media/HD/github/kbkpbot/v
|VMODULES            |OK, value: /home/mars/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/mars/v/bug
|                    |
|Git version         |git version 2.43.0
|V git status        |weekly.2025.10-55-g50b716b8-dirty (4 commit(s) behind V master)
|.git/config present |true
|                    |
|cc version          |cc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|gcc version         |gcc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|clang version       |Ubuntu clang version 18.1.3 (1ubuntu1)
|tcc version         |tcc version 0.9.28rc 2024-07-31 HEAD@1cee0908 (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 0134e9b9-dirty
|emcc version        |N/A
|glibc version       |ldd (Ubuntu GLIBC 2.39-0ubuntu8.4) 2.39

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/checker/assign.v
+++ b/vlib/v/checker/assign.v
@@ -329,6 +329,11 @@ fn (mut c Checker) assign_stmt(mut node ast.AssignStmt) {
 		}
 		node.left_types << left_type
 
+		if right is ast.StructInit {
+			right_sym := c.table.sym(right_type)
+			c.check_any_type(right_type, right_sym, right.pos())
+		}
+
 		if left is ast.ParExpr && is_decl {
 			c.error('parentheses are not supported on the left side of `:=`', left.pos())
 		}

--- a/vlib/v/checker/tests/assign_any_err.out
+++ b/vlib/v/checker/tests/assign_any_err.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/assign_any_err.vv:3:22: error: cannot use type `any` here
+    1 | fn g[T](x T) {
+    2 |         for _ in 0 .. 10 {
+    3 |                 _ := any{}
+      |                      ~~~~~
+    4 |         }
+    5 | }

--- a/vlib/v/checker/tests/assign_any_err.out
+++ b/vlib/v/checker/tests/assign_any_err.out
@@ -1,7 +1,7 @@
-vlib/v/checker/tests/assign_any_err.vv:3:22: error: cannot use type `any` here
+vlib/v/checker/tests/assign_any_err.vv:3:8: error: cannot use type `any` here
     1 | fn g[T](x T) {
-    2 |         for _ in 0 .. 10 {
-    3 |                 _ := any{}
-      |                      ~~~~~
-    4 |         }
+    2 |     for _ in 0 .. 10 {
+    3 |         _ := any{}
+      |              ~~~~~
+    4 |     }
     5 | }

--- a/vlib/v/checker/tests/assign_any_err.vv
+++ b/vlib/v/checker/tests/assign_any_err.vv
@@ -1,0 +1,9 @@
+fn g[T](x T) {
+	for _ in 0 .. 10 {
+		_ := any{}
+	}
+}
+
+fn main() {
+	g[int](1)
+}

